akka-http-compatibility-tests/src/main/scala/example/HostConnectionPoolCompat.scala

```diff
@@ -0,0 +1,35 @@
+package example
+
+import akka.http.scaladsl.Http.HostConnectionPool
+
+object HostConnectionPoolCompat {
+  def access(hcp: HostConnectionPool): Unit = {
+    val theSetup = hcp.setup
+    hcp.shutdown()
+
+    hcp match {
+      // This still works because case class matching does not require an unapply method
+      case HostConnectionPool(setup) => require(setup == theSetup)
+    }
+
+    require(hcp.productArity == 1)
+    require(hcp.productElement(0) == hcp.setup)
+    require(hcp.canEqual(hcp))
+    require(hcp.equals(hcp))
+
+    // This one didn't compile even before (private[http] constructor)
+    // new HostConnectionPool(hcp.setup)(null)
+
+    // This line compiles but only by giving null as the PoolGateway parameter which was private before
```

Comment:
I tried to come up with examples that would crash and these do but I think they are not worthwhile to be supported.